src/Surfnet/StepupSelfService/SelfServiceBundle/Service/RemoteVettingService.php

```diff
@@ -0,0 +1,194 @@
+        $this->session->remove(self::TOKEN_SESSION_KEY);
+
+        return $token;
+    }
+
+    /**
+     * @param RemoteVettingTokenDto $token
+     * @param string $identityProviderName
+     * @param string $nameId
+     * @param string $uthnContextClassRef
+     * @return string
+     */
+    public function createAuthnRequest(RemoteVettingTokenDto $token, $identityProviderName)
+    {
+        $this->logger->info('Creating a SAML2 AuthnRequest to send to the IdP');
+
+        $identityProvider = $this->identityProviderFactory->create($identityProviderName);
+        $serviceProvider = $this->serviceProviderFactory->create();
+        $authnRequest = AuthnRequestFactory::createNewRequest($serviceProvider, $identityProvider);
+
+        // Set NameId
+        $authnRequest->setSubject('', Constants::NAMEID_UNSPECIFIED);
+
+        // Set AuthnContextClassRef
+        $authnRequest->setAuthenticationContextClassRef(Constants::AC_UNSPECIFIED);
+
+        // Set RequestId to be able to validate response
+        $token->setRequestId($authnRequest->getRequestId());
+        $this->session->set(self::TOKEN_SESSION_KEY, $token);
+
+        // Create redirect response.
+        $query = $authnRequest->buildRequestQuery();
+
+        return sprintf(
+            '%s?%s',
+            $identityProvider->getSsoUrl(),
+            $query
+        );
+    }
+
+    /**
+     * @param Request $request
+     * @param string $identityProviderName
+     * @return RemoteVettingTokenDto
+     */
+    public function handleResponse(Request $request, $identityProviderName)
+    {
+        // Load the registering/authenticating token
+        /** @var RemoteVettingTokenDto $token */
+        $token = $this->session->get(self::TOKEN_SESSION_KEY);
+
+        if (!$token) {
+            throw new InvalidRemoteVettingAuthenticationContextException(
+                'Unable to find active authentication context'
+            );
+        }
+
+        $identityProvider = $this->identityProviderFactory->create($identityProviderName);
+        $serviceProvider = $this->serviceProviderFactory->create();
+
+        $this->logger->info('Process the SAML Response');
+        $assertion = $this->postBinding->processResponse(
+            $request,
+            $identityProvider,
+            $serviceProvider
+        );
+
+        /** @var SubjectConfirmation $subjectConfirmation */
+        $subjectConfirmation = $assertion->getSubjectConfirmation()[0];
+        $requestId = $subjectConfirmation->SubjectConfirmationData->InResponseTo;
+
+        if ($requestId != $token->getRequestId()) {
+            throw new InvalidRemoteVettingResponseException('The received response is not an answer to the sended request');
+        }
+
+
+        // todo: catch assertions and validate them
+        $a = $assertion->getAttributes();
+
+        $this->logger->info('The NameId value matched the email address of the registering/authenticating token');
```

Comment:
This can be removed/updated to our current use case I assume?

Reply:
I didn't bother because I wanted a functional MVP